CHANGELOG.md

```diff
@@ -11,6 +11,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)
+- Add `WithSamplingStrategyFetcher` which sets custom fetcher implementation. (#4045)
```

Comment:
I am curious if you still need this functionality. You mentioned only the problem with panic in the PR description. If you do not need this functionality I would rather not add it and just have the fix for the panic.

Reply:
We use [httpmock](https://github.com/jarcoal/httpmock) which interferes with a default `http.Client`. To fix it we use custom HTTP Clients, e.g. in fetcher implementation. If that's no problem for you, let's leave it public.